backend/src/v2/driver/driver.go

```diff
@@ -2006,7 +2028,253 @@ func resolveUpstreamArtifacts(cfg resolveUpstreamOutputsConfig) (*pipelinespec.A
 	}
 }
 
-// provisionOuutputs prepares output references that will get saved to MLMD.
+func isCollection(cfg resolveUpstreamOutputsConfig, currentTask *metadata.Execution) ([]string, error) {
+	starterTaskQueue := make([]string, 0)
+	parentDAG, err := cfg.mlmd.GetExecution(cfg.ctx, currentTask.GetExecution().GetCustomProperties()["parent_dag_id"].GetIntValue())
+	if err != nil {
+		return nil, err
+	}
+	resolveCollection, iterations := getParallelForIterationCount(currentTask, parentDAG)
+
+	glog.V(4).Infof("resolveCollection: %v", resolveCollection)
+	glog.V(4).Infof("iterations: %v", iterations)
+	if resolveCollection {
+		for i := range iterations {
+			starterTaskName := metadata.UpdateTaskNameWithDagID(currentTask.TaskName(), currentTask.GetID())
+			starterTaskName = metadata.GetParallelForTaskName(starterTaskName, i)
+			starterTaskQueue = append(starterTaskQueue, starterTaskName)
+		}
+	}
+	return starterTaskQueue, nil
+}
+
+// In order to properly perform artifact/parameter collection across a DAG/tree
```

Comment:
How did v1 handle this? Just relied on AWF to the do the heavy lifting? I think that maybe, given the complexity of this function, it might make sense to expand the docstring and maybe even add parameter descriptions.